oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/CachingCommitValueResolver.java

```diff
@@ -148,7 +149,13 @@ public String resolve(@NotNull Revision changeRevision,
             // only cache committed states
             // e.g. branch commits may be merged later and
             // the commit value will change
-            commitValueCache.put(changeRevision, value);
+            try {
+                commitValueCache.put(changeRevision, value);
+            } catch (RuntimeException re) {
+                LoggerFactory.getLogger(getClass()).error("resolve: RuntimeException with " +
+                        changeRevision + " - " + value, re);
+                throw re;
```

Comment:
How about turning this into a warn message and *not* re-throw the exception? I understand the issue is probably with Guava because both key and value are non-null. Guavas LocalCache.put() even checks this.

Reply:
Good point, it probably doesn't matter if that `put` fails - so a rethrow isnt strictly necessary.
Changed that now [here](https://github.com/apache/jackrabbit-oak/pull/449/commits/5a7b331af20345d2e75a068fc183cd1b0c1886c8)